server/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -2024,11 +2024,20 @@ public void testSupportsParallelCollection() {
         if (searchSourceBuilder != null && randomBoolean()) {
             searchSourceBuilder.aggregation(new TermsAggregationBuilder("terms"));
         }
-        assertTrue(SearchService.supportsParallelCollection(ResultsType.DFS, searchSourceBuilder));
+        assertTrue(SearchService.supportsParallelCollection(ResultsType.DFS, searchSourceBuilder, true));
         assertFalse(
             SearchService.supportsParallelCollection(
                 randomFrom(randomFrom(ResultsType.QUERY, ResultsType.NONE, ResultsType.FETCH)),
-                searchSourceBuilder
+                searchSourceBuilder,
+                false
+            )
+        );
+        assertFalse(
+            // TODO: until https://github.com/elastic/elasticsearch/pull/98425 is merged, this should always return false.
+            SearchService.supportsParallelCollection(
+                randomFrom(randomFrom(ResultsType.QUERY, ResultsType.NONE, ResultsType.FETCH)),
+                searchSourceBuilder,
+                true
```

Comment:
I would add a test to verify default value / updating the setting and its effect. We have a similar one already for disabling the search worker threads.

Reply:
Good point!  I added `testEnableQueryPhaseParallelCollection()` to this effect.